tests/test_caching.py

```diff
@@ -51,45 +58,6 @@ def test_cache_id(tmp_path):
     assert linker_uid == random_uid
 
 
-def test_materialising_works():
```

Comment:
I don't think these tests are needed now we're explicitly using a fresh pipeline for everything - there's no longer the potential for accidentally forgetting to reset the pipeline etc. and any future errors should be much clearer if they do occur